docs/examples/s2n_recv.c

```diff
@@ -21,11 +21,24 @@ int s2n_example_recv(struct s2n_connection *conn, uint8_t *buffer, size_t buffer
     int bytes_read = 0;
     while (bytes_read < buffer_size) {
         int r = s2n_recv(conn, buffer + bytes_read, buffer_size - bytes_read, &blocked);
-        if (r == 0) {
-            break;
-        } else if (r > 0) {
+        if (r > 0) {
+            /* `r` bytes were successfully received. Update `bytes_read` so the next `s2n_recv()`
+             * call writes into the buffer with the correct offset.
+             */
             bytes_read += r;
-        } else if (s2n_error_get_type(s2n_errno) != S2N_ERR_T_BLOCKED) {
+        } else if (r == 0) {
+            /* The connection was closed. No more data will be received, so bail out of the loop. */
+            break;
+        } else if (s2n_error_get_type(s2n_errno) == S2N_ERR_T_BLOCKED) {
+            /* The return of `s2n_recv()` indicates an error. If this error is `S2N_ERR_T_BLOCKED`,
+             * the socket is blocked waiting to receive more data from the peer. In a typical
+             * non-blocking environment, `poll()` or `select()` would be used re-enter
+             * `s2n_example_recv()` and call `s2n_recv()` again after the socket has more data
+             * available.
+             */
+            continue;
```

Comment:
Does this add any additional clarity? (We could make the same change to send)

```suggestion
        } else if (r < 0 && s2n_error_get_type(s2n_errno) == S2N_ERR_T_BLOCKED) {
            /* The return of `s2n_recv()` indicates an error. If this error is `S2N_ERR_T_BLOCKED`,
             * the socket is blocked waiting to receive more data from the peer.
             *
             * In a typical non-blocking environment, `poll()` or `select()` would be used re-enter
             * `s2n_example_recv()` and call `s2n_recv()` again after the socket has more data
             * available. But this example just busy-waits, so we immediately call `s2n_recv()` again.
             */
            continue;
```

Reply:
Added, thanks!